Add option to configure the config file path

### Discussed in https://github.com/HoudiniGraphql/houdini/discussions/300

<div type='discussions-op-text'>

<sup>Originally posted by **tadeokondrak** May 23, 2022</sup>
I'm working on a project where the frontend isn't at the root of the workspace. Most things work fine, but the Svelte language server needs to run the preprocessor to find errors, and since it runs in the root of the workspace, it shows an error in many files:

`Cannot find module '.../repo/houdini.config.js' imported from .../repo/frontend/houdini/build/preprocess-esm/index.js`

The file structure is something like:
```
repo/
repo/frontend/svelte.config.js
repo/frontend/houdini.config.js
repo/frontend/$houdini/...
repo/backend/...
```

I'm not sure what the best way to fix this is. Should the language server be running the preprocessors in the directory with svelte.config.js, or can Houdini do something to look for the config differently?</div>

--- a/.changeset/lazy-kings-watch.md
+++ b/.changeset/lazy-kings-watch.md
@@ -1,0 +1,5 @@
+---
+'houdini': patch
+---
+
+Add a `configFile` parameter to the preprocessor so users can specify where to find their `houdini.config.js` file

--- a/src/common/config.ts
+++ b/src/common/config.ts
@@ -546,13 +546,16 @@ async function loadSchemaFile(schemaPath: string): Promise<graphql.GraphQLSchema
 }
 
 // get the project's current configuration
-export async function getConfig(extraConfig?: Partial<ConfigFile>): Promise<Config> {
+export async function getConfig({
+	configFile,
+	...extraConfig
+}: { configFile?: string } & Partial<ConfigFile>): Promise<Config> {
 	if (_config) {
 		return _config
 	}
 
 	// add the filepath and save the result
-	const configPath = DEFAULT_CONFIG_PATH
+	const configPath = configFile || DEFAULT_CONFIG_PATH
 	const config = await readConfigFile(configPath)
 
 	// look up the schema

--- a/src/common/config.ts
+++ b/src/common/config.ts
@@ -549,7 +549,7 @@ async function loadSchemaFile(schemaPath: string): Promise<graphql.GraphQLSchema
 export async function getConfig({
 	configFile,
 	...extraConfig
-}: { configFile?: string } & Partial<ConfigFile>): Promise<Config> {
+}: { configFile?: string } & Partial<ConfigFile> = {}): Promise<Config> {
 	if (_config) {
 		return _config
 	}

--- a/src/preprocess/index.ts
+++ b/src/preprocess/index.ts
@@ -3,8 +3,16 @@ import { getConfig } from '../common'
 import { ConfigFile } from '../runtime'
 import applyTransforms from './transforms'
 
-// the main entry point for the preprocessor
-export default function houdiniPreprocessor(extraConfig?: Partial<ConfigFile>) {
+/**
+ * The houdini processor automates a lot of boilerplate to make inline documents
+ * work.
+ *
+ * It takes the same configuration values as the houdini config file as well as an
+ * optional `configFile` parameter to specify the path to use to find houdini.config.js
+ */
+export default function houdiniPreprocessor(
+	extraConfig: { configFile?: string } & Partial<ConfigFile>
+) {
 	return {
 		async markup({ content, filename }: { content: string; filename: string }) {
 			// grab the config

